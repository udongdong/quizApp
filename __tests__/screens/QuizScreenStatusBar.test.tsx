import 'react-native';
import React from 'react';
import {
  QuizScreenStatusBar,
  QuizScreenStatusBarProps,
} from '../../src/screens/quiz/quizScreen/QuizScreenStatusBar';

import {render} from '@testing-library/react-native';

function getComponent(props: QuizScreenStatusBarProps): React.JSX.Element {
  return <QuizScreenStatusBar {...props} />;
}

describe('QuizScreenStatusBar test', () => {
  test('render QuizScreenStatusBar', () => {
    const props: QuizScreenStatusBarProps = {
      total: 0,
      current: 0,
      time: 100,
    };

    const rendered = render(getComponent(props));

    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();

    expect(rendered.getByText('01:40')).toBeDefined();
  });
});
