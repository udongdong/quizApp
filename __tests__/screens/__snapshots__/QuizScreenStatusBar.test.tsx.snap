// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizScreenStatusBar test render QuizScreenStatusBar 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "padding": 15,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#ddd",
          "borderRadius": 5,
          "height": 10,
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#4caf50",
            "borderRadius": 5,
            "height": "100%",
            "width": "NaN%",
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      {
        "padding": 10,
        "width": "19%",
      }
    }
  >
    01:40
  </Text>
</View>
`;
